.github/workflows/apple-perf.yml

```diff
@@ -185,14 +186,25 @@ jobs:
           name: ios-models
           path: ${{ runner.temp }}/artifacts/
 
-      - name: Verify the models
+      - name: Verify models
+        id: verify-models
         shell: bash
         working-directory: ${{ runner.temp }}/artifacts/
         run: |
           ls -lah ./
 
+          # Check if the directory contains any model
+          if [ -z "$(ls -A .)" ]; then
+            echo "No models found."
+            echo "::set-output name=models-found::false"
+          else
+            echo "Models found."
+            echo "::set-output name=models-found::true"
+          fi
+
       - name: Upload the models to S3
         uses: seemethere/upload-artifact-s3@v5
+        if: steps.verify-models.outputs.models-found == 'true'  # Upload only if models were found
```

Comment:
I suspect that you might not need to change anything in `upload-models` job besides setting `if :always()`.  The upload step here already has `if-no-files-found: ignore` which does what it says by ignoring if there is no file

Reply:
Cool! Good to know.